opensearch/src/main/java/org/opensearch/sql/opensearch/request/AggregateAnalyzer.java

```diff
@@ -417,7 +417,7 @@ private static Pair<AggregationBuilder, MetricParser> createRegularAggregation(
         } else {
           yield Pair.of(
               AggregationBuilders.topHits(aggFieldName)
-                  .fetchSource(helper.inferNamedField(args.getFirst()).getRootName(), null)
```

Comment:
I think we should prefer fetch fields API. This will break for certain field types, e.g., text, nested. Please double check.